python/tvm/relay/backend/contrib/ethosu/tir_to_cs_translator.py

```diff
@@ -254,15 +255,19 @@ def extract_param_base_addresses(mod, buffer_info, scratch_region_map) -> List[u
     assert len(mod.functions.items()) == 1
     primfunc = mod.functions.items()[0][1]
 
+    buffer_map = tir_utils.collect_buffer_map(primfunc.body)
+
     base_addresses = list()
     idx = 0
+
     for param in primfunc.params:
         # constants are pooled together and handled specially
         # this will change after tir.allocate_const.
         # For now, we are skipping generating buffer addresses here
         if buffer_info[param].btype == BufferType.constant:
             continue
-        buffer = primfunc.buffer_map[param]
+
```

Comment:
I've just chatted with @ekalda around this and we're not sure why the extra buffer appears, it appears to be an unused intermediary buffer and for the tests I ran this fixes it:

```suggestion
    if param not in buffer_map:
        base_addresses.append(
            util.BaseAddress(
                param.name.replace("-", "_"),
                idx,
                _get_region(buffer_info[param].btype, param, scratch_region_map),
                0,
            )
        )
        idx += 1
        continue
```

I can tidy this loop up afterwards :smile_cat:

Reply:
That works for me locally, on to the CI!  Thank you on the ethos-u debugging, and it helps quite a bit.